### PR TITLE
Docs: Add details for format-js to @wordpress/scripts package

### PR DESCRIPTION
**Pull request by gziolo**
## Description
Follow-up for #18048.

This PR adds all the essential details about the new `format-js` script added to `@wordpress/scripts` package.

I added more changes to clarify some recent changes introduced in `@wordpress/scripts` as well:
- the required version of node and npm in `check-engines` script
- a note about the `update-packages` script in the section about self-upgrade of `@wordpress/scripts` when declared as a dependency in the project

In addition, it applies some modifications to `@wordpress/eslint-plugin` to reflect changes applied as part of the efforts.

## Checklist:
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

File: packages/eslint-plugin/README.md [hunk 1]
```diff
@@ -32,7 +34,6 @@ Alternatively, you can opt-in to only the more granular rulesets offered by the
 - `es5`
 - `esnext`
 - `jsdoc`
-- `jshint` (legacy)
```
Comment (gziolo) [inline packages/eslint-plugin/README.md]: It's covered separately at the end of the file. It should be less exposed.

File: packages/eslint-plugin/CHANGELOG.md [hunk 2]
```diff
@@ -2,8 +2,8 @@
 
 ### New Features
 
-- The `recommended` config no longer enables rules that check code formatting (whitespace, indenting, etc.) and that could conflict with Prettier.
-- There is a new `recommended-with-formatting` config that has the code formatting rules still enabled, for projects that want to opt out from Prettier and continue checking code formatting with ESLint.
+- The `recommended` rulest no longer enables rules that check code formatting (whitespace, indenting, etc.) and that could conflict with Prettier.
```
Comment (mkaz) [inline packages/eslint-plugin/CHANGELOG.md]: Minor typo

```suggestion
- The `recommended` ruleset no longer enables rules that check code formatting (whitespace, indenting, etc.) and that could conflict with Prettier.
```

**Review by mkaz — APPROVED** [review]
One small change but looks great. :+1:

I had queued up a similar PR in #19766 but I'll close mine, yours is more complete

**Comment by gziolo** [comment]
> I had queued up a similar PR in #19766 but I'll close mine, yours is more complete

I totally forgot about it, I should build upon it. Sorry about 😅

**Pull request merged**